openeo/rest/datacube.py

```diff
@@ -1829,6 +1830,71 @@ def validate(self) -> List[dict]:
     def tiled_viewing_service(self, type: str, **kwargs) -> Service:
         return self._connection.create_service(self.flat_graph(), type=type, **kwargs)
 
+    def _get_spatial_extent_from_load_collection(self):
+        pg = self.flat_graph()
+        for node in pg:
+            if pg[node]["process_id"] == "load_collection":
+                if "spatial_extent" in pg[node]["arguments"] and all(
+                    cd
+                    for cd in pg[node]["arguments"]["spatial_extent"]
+                    for cd in ["east", "west", "south", "north"]
+                ):
+                    return pg[node]["arguments"]["spatial_extent"]
+        return None
+
+
+    def preview(
+        self,
+        center: Union[Iterable, None] = None,
+        zoom: Union[int, None] = None,
+    ) -> Tuple[Map, Service]:
+        """
+        Creates a service with the process graph and displays a map widget. Only supports XYZ.
+
+        :param center: (optional) Map center. Default is (0,0).
+        :param zoom: (optional) Zoom level of the map. Default is 1.
+
+        :return: ipyleaflet Map object and the displayed Service
+        """
+        if "XYZ" not in self.connection.list_service_types():
+            raise OpenEoClientException(f"Backend does not support service type 'XYZ'.")
+
+        service = self.tiled_viewing_service("XYZ")
+        service_metadata = service.describe_service()
+
+        m = Map(
+            center=center or (0, 0),
+            zoom=zoom or 1,
+            scroll_wheel_zoom=True,
+            basemap=basemaps.OpenStreetMap.Mapnik,
+        )
+        service_layer = TileLayer(url=service_metadata["url"])
+        m.add_layer(service_layer)
+
+        if center is None and zoom is None:
+            spatial_extent = self._get_spatial_extent_from_load_collection()
+            if spatial_extent is not None:
+                m.fit_bounds(
+                    [
+                        [spatial_extent["south"], spatial_extent["west"]],
+                        [spatial_extent["north"], spatial_extent["east"]],
+                    ]
+                )
+
+        try:
+            # Check we are inside Jupyer Notebook
```

Comment:
Maybe I'm missing some context, but is there another context (other than Jupyter notebooks), where this `preview()` method makes sense?  If you can not `display()`  the `Map`, is it useful to construct it in the first place?